hardhat.config.ts

```diff
@@ -15,6 +16,182 @@ const accounts: HDAccountsUserConfig = {
+      taskArgs.useDefender &&
+      !SUPPORTED_NETWORKS.includes(hre.network.name)
+    ) {
+      throw Error(
+        `--use-defender only supports networks: ${SUPPORTED_NETWORKS}`,
+      )
+    }
+
+    const ownerAddress =
+      taskArgs.ownerAddress || (await hre.ethers.getSigners())[0].address
+
+    await deployContract(hre, 'Registry', [ownerAddress, ONE_DAY], {
+      useDefender: taskArgs.useDefender,
+      defenderDeploySalt: taskArgs.defenderDeploySalt,
+    })
+  })
+
+task('deploy-reward-pool', 'Deploy RewardPool contract')
+  .addParam('poolToken', 'Address of the token used for rewards')
+  .addOptionalParam('managerAddress', 'Address that will have MANAGER_ROLE')
+  .addOptionalParam('rewardFunction', 'Identifier of the reward function')
+  .addOptionalParam(
+    'timelock',
+    'Timestamp when manager withdrawals will be allowed',
+    0,
+    types.int,
+  )
+  .addOptionalParam('ownerAddress', 'Address to use as owner')
+  .addFlag('useDefender', 'Deploy using OpenZeppelin Defender')
+  .addOptionalParam('defenderDeploySalt', 'Salt to use for CREATE2 deployments')
+  .setAction(async (taskArgs, hre) => {
+    if (
+      taskArgs.useDefender &&
+      !SUPPORTED_NETWORKS.includes(hre.network.name)
+    ) {
+      throw Error(
+        `--use-defender only supports networks: ${SUPPORTED_NETWORKS}`,
+      )
+    }
+
+    const ownerAddress =
+      taskArgs.ownerAddress || (await hre.ethers.getSigners())[0].address
+
+    const managerAddress = taskArgs.managerAddress || ownerAddress
+
+    const rewardFunctionId = ethers.zeroPadValue(
+      taskArgs.rewardFunction || '0x00',
+      32,
+    )
+
+    await deployContract(
+      hre,
+      'RewardPool',
+      [
+        taskArgs.poolToken,
+        rewardFunctionId,
+        ownerAddress,
+        ONE_DAY,
+        managerAddress,
+        taskArgs.timelock,
+      ],
+      {
+        isUpgradeable: true,
+        useDefender: taskArgs.useDefender,
+        defenderDeploySalt: taskArgs.defenderDeploySalt,
+      },
+    )
+  })
+
+task('upgrade-reward-pool', 'Upgrade RewardPool contract')
+  .addParam('proxyAddress', 'Address of the token used for rewards')
+  .addFlag('useDefender', 'Deploy using OpenZeppelin Defender')
+  .addOptionalParam('defenderDeploySalt', 'Salt to use for CREATE2 deployments')
+  .setAction(async (taskArgs, hre) => {
+    if (
+      taskArgs.useDefender &&
+      !SUPPORTED_NETWORKS.includes(hre.network.name)
+    ) {
+      throw Error(
+        `--use-defender only supports networks: ${SUPPORTED_NETWORKS}`,
+      )
+    }
+
+    await upgradeContract(hre, 'RewardPool', taskArgs.proxyAddress, {
+      useDefender: taskArgs.useDefender,
+      defenderDeploySalt: taskArgs.defenderDeploySalt,
+    })
+  })
+
+task('deploy-mock-token', 'Deploy mock ERC-20 token').setAction(
+  async (_, hre) => {
+    await deployContract(hre, 'MockERC20', ['Mock ERC20', 'MOCK'])
+  },
+)
+
+// Contract deployment helper
+async function deployContract(
```

Comment:
Could we print the constructor args too?
This helps ensure it's being deployed with the expected values.